src/interactive/AccordionPanelGroup.jsx

```diff
@@ -5,12 +5,25 @@ import cx from 'classnames';
 export const ACCORDIONPANELGROUP_CLASS = 'accordionPanelGroup';
 
 /**
- * Determines if the passed in value is a primitive type
- * @param {any} value
- * @return {boolean} True if value is a primitive type
+ * @description takes AccordionPanelGroup and AccordionPanel data and returns an array of new panel `isOpen` states
+ * @param {Boolean} isMultiselect
+ * @param {Array} statesList
+ * @param {Object} panelData
+ * @returns {Array} newPanelStatesList - an array of each panel's `isOpen` prop value
  */
-export const isPrimitive = value =>
-	['object', 'function'].indexOf(typeof value) === -1;
+export const getNewPanelState = (isMultiselect, statesList, panelData) => {
```

Comment:
optional code style thing: this could return immediately instead of using a function statement block...

```
export const getNewPanelState = (isMultiselect, statesList, panelData) => statesList
	.map((panelState, i) => {
		...
	});
```